### PR TITLE
Status Page E2E spec

### PR DESCRIPTION
**Pull request by simshaun**
- [x] I have read and understand the pull request rules.

# Description

Adds an E2E test for Status pages.

This E2E test ensures a status page can be created and edited. It tests most of the editable fields.

Items not yet being tested:
- Show certificate expiry
- Domain names _(haven't thought about how to go about testing this)_

## Type of change

- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [x] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

None

### CODE REVIEW

File: config/playwright.config.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ export default defineConfig({
 
     // Run your local dev server before starting the tests.
     webServer: {
-        command: `node extra/remove-playwright-test-data.js && node server/server.js --port=${port} --data-dir=./data/playwright-test`,
+        command: `node extra/remove-playwright-test-data.js && cross-env NODE_ENV=development node server/server.js --port=${port} --data-dir=./data/playwright-test`,
```
Comment (CommanderStorm) [inline config/playwright.config.js:61]: Is this change intentional?

File: test/e2e/specs/example.spec.js [hunk 2]
```diff
@@ -13,25 +13,26 @@ test.describe("Example Spec", () => {
         await screenshot(testInfo, page);
     });
 
-    test("change display timezone", async ({ page }, testInfo) => {
```
Comment (CommanderStorm) [inline test/e2e/specs/example.spec.js:16]: Is removing the Timezone testcase intentional?

File: test/e2e/specs/status-page.spec.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+        await expect(page.locator("body")).toHaveClass(theme);
+        expect(await page.locator("head").innerHTML()).toContain(googleAnalyticsId);
+
+        const backgroundColor = await page.evaluate(() => window.getComputedStyle(document.body).backgroundColor);
+        expect(backgroundColor).toEqual("rgb(0, 128, 128)");
+
+        await screenshot(testInfo, page);
+
+        // Flip the "Show Tags" and "Show Powered By" switches:
```
Comment (CommanderStorm) [inline test/e2e/specs/status-page.spec.js:110]: I am not a testing expert.
Do you think it could make sense to move the setup code above to `beforeEach`?

**Review by CommanderStorm — COMMENTED** [review]
Awesome, your help is much appreciated ♥️

File: test/e2e/specs/example.spec.js (hunk 2 above)
Comment (simshaun) [reply test/e2e/specs/example.spec.js:16]: Yes. I discovered my example spec that was intended to also test resetting the database was actually flawed. The database reset wasn't occurring (discovered while writing this Status Page spec) because the test server isn't started in a dev environment to expose the e2e routes. The example test was succeeding when it wasn't supposed to because (if I determined correctly) localStorage saving the timezone. So I changed the test to something that isn't prone to such issue.

File: config/playwright.config.js (hunk 1 above)
Comment (simshaun) [reply config/playwright.config.js:61]: See below comment.

File: test/e2e/specs/status-page.spec.js (hunk 3 above)
Comment (simshaun) [reply test/e2e/specs/status-page.spec.js:110]: I think it would be better to do when & if more cases are added to the spec.

**Review by CommanderStorm — APPROVED** [review]
Thanks 🙏
